src/system_identity/client_input.rs

```diff
@@ -0,0 +1,20 @@
+use super::environment::SystemIdentityCreationEnvironment;
+
+/// Represents the input data required to create a System Identity.
+#[derive(Debug, Clone, PartialEq)]
+pub struct SystemIdentityCreationMetadata {
+    pub name: String,
+    pub organization_id: String,
+    pub client_id: String,
+    pub auth_method: AuthMethod,
+    pub environment: SystemIdentityCreationEnvironment,
+}
+
+type ClientSecret = String; // TODO type better
+type PrivateKeyPem = Vec<u8>; // TODO Type better
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum AuthMethod {
+    ClientSecret(ClientSecret),         // L1 method
+    FromLocalPrivateKey(PrivateKeyPem), // L2 method
```

Comment:
I know you still have to improve types, so I will just throw an idea that maybe you already had in mind.

Authentication has nothing to do with how a token is signed. I think we need two axis here. One for the `AuthMethod` and one for `SigningTool`. Maybe it makes things harder and we prefer having it encoded here. I don't know.

Reply:
Yeah, makes sense. We have the IAMClient and internally, it can use several authentication methods. Okay. I'll wait the refactor.